canonical/vanilla-framework
# Sidenav toggle button triggers on spacebar press

## Done

Changes all uses of `a.p-side-navigation__toggle` and `a.p-side-navigation__toggle--in-drawer` to `<button>` elements and removed `href` attribute

Fixes #4559, [WD-11071](https://warthogs.atlassian.net/browse/WD-11071)

## QA

- Open a sidenav
    - [Accordion / Dark](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/accordion-dark)
    - [Accordion](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/accordion)
    - [Side navigation dark](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/dark)
    - [Side navigation default](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/default)
    - [Side navigation docs](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/docs)
    - [Expandable dark](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/expandable-dark)
    - [Expandable](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/expandable)
    - [Icons](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/icons)
    - [Paper](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/paper)
    - [Raw HTML](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/raw-html)
    - [Sticky](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/sticky)
    - [Vanilla Docs](https://vanilla-framework-5084.demos.haus/docs)
- Small / Medium breakpoints:
    - Verify that the side-navigation toggle button visually appears as it does in the current production release (button styling)
    - Verify that you can open/close the side-navigation using all of the following methods:
        - Click / tap
        - Enter key
        - Space key (new functionality)
    - Verify that the open/close side-navigation button is read to you by a screen reader such as [ChromeVox](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn).
- All breakpoints:
    - Verify that the side-navigation content/structure is read to you by a screen reader such as [ChromeVox](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn).

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
![Screenshot 2024-05-07 at 1 44 28 PM](https://github.com/canonical/vanilla-framework/assets/46915153/160a2c13-0d60-4904-934f-c426dd881377)



[WD-11071]: https://warthogs.atlassian.net/browse/WD-11071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
